src/lib.rs

```diff
@@ -192,21 +181,42 @@ impl Pixels {
         let mut encoder = self
             .device
             .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
-        for renderer in self.renderers.iter_mut() {
-            renderer.resize(&mut encoder, width, height);
-        }
-
-        self.queue.borrow_mut().submit(&[encoder.finish()]);
+        self.scaling_renderer
+            .resize(&mut self.device, &mut encoder, width, height);
+        self.queue.submit(&[encoder.finish()]);
     }
 
-    /// Draw this pixel buffer to the configured [`SurfaceTexture`].
-    ///
-    /// This executes all render passes in sequence. See [`RenderPass`].
+    /// Provides a [`wgpu::CommandEncoder`], a [`wgpu::TextureView`] from the swapchain which you can use to render to the screen, and the default [`ScalingRenderer`].
     ///
     /// # Errors
     ///
     /// Returns an error when [`wgpu::SwapChain::get_next_texture`] times out.
-    pub fn render(&mut self) -> Result<(), Error> {
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use pixels::Pixels;
+    /// # let surface = wgpu::Surface::create(&pixels_mocks::RWH);
+    /// # let surface_texture = pixels::SurfaceTexture::new(1024, 768, surface);
+    /// let mut pixels = Pixels::new(320, 240, surface_texture)?;
+    ///
+    /// // Clear the pixel buffer
+    /// let frame = pixels.get_frame();
+    /// for pixel in frame.chunks_exact_mut(4) {
+    ///     pixel[0] = 0x00; // R
+    ///     pixel[1] = 0x00; // G
+    ///     pixel[2] = 0x00; // B
+    ///     pixel[3] = 0xff; // A
+    /// }
+    ///
+    /// // Draw it to the `SurfaceTexture`
+    /// pixels.render(|encoder, render_target, scaling_renderer| {
+    ///    scaling_renderer.render(encoder, render_target);
+    /// });
+    pub fn render<F: FnOnce(&mut wgpu::CommandEncoder, &wgpu::TextureView, &ScalingRenderer)>(
```

Comment:
I'm thinking this `FnOnce` type can be named and defined elsewhere to increase the readability of this function signature.

Reply:
Sure.